app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Database\QueryException;
+use Illuminate\Http\Request;
+use App\User;
+
+class UserController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $users = User::all();
+        return response()->json(['status' => 'success', 'data' => ['users' => $users]]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
+            'uid' => 'required|unique:users|max:127',
+            'gtid' => 'required|unique:users|max:10',
+            'slack_id' => 'unique:users|max:21',
+            'gt_email' => 'required|unique:users|max:255',
+            'personal_email' => 'unique:users|max:255',
+            'first_name' => 'required|max:127',
+            'middle_name' => 'max:127',
+            'last_name' => 'required|max:127',
+            'preferred_name' => 'max:127',
+            'phone' => 'max:15',
+            'emergency_contact_name' => 'max:255',
+            'emergency_contact_phone' => 'max:15',
+            'join_semester' => 'max:6',
+            'graduation_semester' => 'max:6',
+            'shirt_size' => 'in:s,m,l,xl,xxl,xxxl',
+            'polo_size' => 'in:s,m,l,xl,xxl,xxxl',
+        ]);
+
+        try {
+            $user = User::create($request->all());
+        } catch (QueryException $e) {
+            $errorMessage = $e->errorInfo[2];
+            return response()->json(['status' => 'error', 'message' => $errorMessage], 500);
+        }
+
+        if (is_numeric($user->id)) {
```

Comment:
I would make GTID or UID options, determined based on int or string and make id an optional query param. The db ID is unlikely to every be used off the server, since we're enforcing UID and GTID as unique keys. I would also be comfortable falling back to Id if the number format is not a GTID formatted number.